x-pack/solutions/search/plugins/search_synonyms/server/lib/delete_synonym_rule.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient } from '@kbn/core/server';
+
+export const deleteSynonymRule = async (
+  client: ElasticsearchClient,
+  synonymsSetId: string,
+  ruleId: string
+) => {
+  return await client.synonyms.deleteSynonymRule({ set_id: synonymsSetId, rule_id: ruleId });
```

Comment:
since this is already an `async` function this `await` is redundant. you can just return the promise without waiting for it resolve if you're not going to handle the exception in this function.